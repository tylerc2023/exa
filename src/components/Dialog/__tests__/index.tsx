import React from "react";
import renderer from "react-test-renderer";
import Component from "..";

jest.useFakeTimers();
describe("Dialog component", () => {
  it("renders correctly with defaults and hidden", () => {
    const dom = renderer.create(<Component title="bob" />);
    expect(dom.toJSON()).toMatchSnapshot();
  });

  it("renders correctly with defaults and visible", () => {
    const dom = renderer.create(<Component title="bob" visible />);
    expect(dom.toJSON()).toMatchSnapshot();
  });

  it("renders correctly with props", () => {
    const dom = renderer.create(
      <Component
        visible
        title="nope"
        message="yep"
        alert
        confirmButtonText="dog"
        cancelButtonText="bird"
        backgroundClose
        onConfirmButtonPress={jest.fn()}
        onCancelButtonPress={jest.fn()}
      />
    );
    expect(dom.toJSON()).toMatchSnapshot();
  });

  it("renders alert to show then hide", () => {
    const callback = jest.fn();
    const dom = renderer.create(
      <Component alert title="bob" onCancelButtonPress={callback} />
    );
    const disappearAlert = jest.spyOn(dom.root.instance, "disappearAlert");
    const clearTimeouts = jest.spyOn(dom.root.instance, "clearTimeouts");
    dom.update(
      <Component visible alert title="bob" onCancelButtonPress={callback} />
    );
    expect(disappearAlert).toHaveBeenCalled();
    expect(callback).not.toHaveBeenCalled();
    jest.runAllTimers();
    expect(callback).toHaveBeenCalled();
    expect(clearTimeouts).toHaveBeenCalled();
  });

  it("renders dialog to show single button in center", () => {
    const dom = renderer.create(
      <Component alert title="bob" onCancelButtonPress={() => undefined} />
    );
    expect(dom.toJSON()).toMatchSnapshot();
  });

  it("renders dialog to show double buttons beside each other", () => {
    const dom = renderer.create(
      <Component
        alert
        title="bob"
        onCancelButtonPress={() => undefined}
        onConfirmButtonPress={() => undefined}
      />
    );
    expect(dom.toJSON()).toMatchSnapshot();
  });

  it("renders alert to show then hide without onBackgroundPress", () => {
    const dom = renderer.create(<Component alert title="bob" />);
    const disappearAlert = jest.spyOn(dom.root.instance, "disappearAlert");
    const clearTimeouts = jest.spyOn(dom.root.instance, "clearTimeouts");
    dom.update(<Component visible alert title="bob" />);
    expect(disappearAlert).toHaveBeenCalled();
    jest.runAllTimers();
    expect(clearTimeouts).toHaveBeenCalled();
  });

  it("clears timeouts when disappearing", () => {
    const dom = renderer.create(<Component visible title="bob" />);
    const clearTimeouts = jest.spyOn(dom.root.instance, "clearTimeouts");
    dom.update(<Component title="bob" />);
    expect(clearTimeouts).toHaveBeenCalled();
  });

  it("clears timeouts when un mounting", () => {
    const dom = renderer.create(<Component visible title="bob" />);
    const clearTimeouts = jest.spyOn(dom.root.instance, "clearTimeouts");
    dom.unmount();
    expect(clearTimeouts).toHaveBeenCalled();
  });
});
