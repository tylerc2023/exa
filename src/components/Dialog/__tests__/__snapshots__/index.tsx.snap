// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog component renders correctly with defaults and hidden 1`] = `
<Modal
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "hsla(0, 0%, 0%, 0.5)",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": 32,
      }
    }
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "hsl(0, 100%, 100%)",
          "borderRadius": 16,
          "marginHorizontal": 60,
          "maxWidth": 500,
          "paddingBottom": 36,
          "paddingHorizontal": 36,
          "paddingTop": 52,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 24,
              "fontWeight": "300",
              "letterSpacing": 0,
            },
            undefined,
            undefined,
            Object {
              "paddingBottom": 12,
            },
          ]
        }
      >
        bob
      </Text>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingTop": 52,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</Modal>
`;

exports[`Dialog component renders correctly with defaults and visible 1`] = `
<Modal
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "hsla(0, 0%, 0%, 0.5)",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": 32,
      }
    }
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "hsl(0, 100%, 100%)",
          "borderRadius": 16,
          "marginHorizontal": 60,
          "maxWidth": 500,
          "paddingBottom": 36,
          "paddingHorizontal": 36,
          "paddingTop": 52,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 24,
              "fontWeight": "300",
              "letterSpacing": 0,
            },
            undefined,
            undefined,
            Object {
              "paddingBottom": 12,
            },
          ]
        }
      >
        bob
      </Text>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingTop": 52,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</Modal>
`;

exports[`Dialog component renders correctly with props 1`] = `
<Modal
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "hsla(0, 0%, 0%, 0.5)",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": 32,
      }
    }
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "hsl(0, 100%, 100%)",
          "borderRadius": 16,
          "marginHorizontal": 60,
          "maxWidth": 500,
          "paddingBottom": 36,
          "paddingHorizontal": 36,
          "paddingTop": 52,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 24,
              "fontWeight": "300",
              "letterSpacing": 0,
            },
            undefined,
            undefined,
            Object {
              "paddingBottom": 12,
            },
          ]
        }
      >
        nope
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontSize": 14,
              "fontWeight": "300",
              "letterSpacing": 0.25,
            },
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        yep
      </Text>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingTop": 52,
            },
            Object {
              "display": "none",
            },
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderWidth": 1,
              "color": "hsl(211, 100%, 50%)",
              "flex": 1,
              "flexDirection": "row",
              "height": 36,
              "justifyContent": "center",
              "marginRight": 8,
              "opacity": 1,
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 14,
                  "fontWeight": "600",
                  "letterSpacing": 0.75,
                },
                Object {
                  "textAlign": "center",
                },
                undefined,
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "hsl(211, 100%, 50%)",
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            BIRD
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderWidth": 1,
              "color": "hsl(211, 100%, 50%)",
              "flex": 1,
              "flexDirection": "row",
              "height": 36,
              "justifyContent": "center",
              "marginLeft": 8,
              "opacity": 1,
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 14,
                  "fontWeight": "600",
                  "letterSpacing": 0.75,
                },
                Object {
                  "textAlign": "center",
                },
                undefined,
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "hsl(211, 100%, 50%)",
                  },
                  undefined,
                  Object {
                    "color": "hsl(208, 7%, 46%)",
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            DOG
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`Dialog component renders dialog to show double buttons beside each other 1`] = `
<Modal
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "hsla(0, 0%, 0%, 0.5)",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": 32,
      }
    }
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "hsl(0, 100%, 100%)",
          "borderRadius": 16,
          "marginHorizontal": 60,
          "maxWidth": 500,
          "paddingBottom": 36,
          "paddingHorizontal": 36,
          "paddingTop": 52,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 24,
              "fontWeight": "300",
              "letterSpacing": 0,
            },
            undefined,
            undefined,
            Object {
              "paddingBottom": 12,
            },
          ]
        }
      >
        bob
      </Text>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingTop": 52,
            },
            Object {
              "display": "none",
            },
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderWidth": 1,
              "color": "hsl(211, 100%, 50%)",
              "flex": 1,
              "flexDirection": "row",
              "height": 36,
              "justifyContent": "center",
              "marginRight": 8,
              "opacity": 1,
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 14,
                  "fontWeight": "600",
                  "letterSpacing": 0.75,
                },
                Object {
                  "textAlign": "center",
                },
                undefined,
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "hsl(211, 100%, 50%)",
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            CANCEL
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderWidth": 1,
              "color": "hsl(211, 100%, 50%)",
              "flex": 1,
              "flexDirection": "row",
              "height": 36,
              "justifyContent": "center",
              "marginLeft": 8,
              "opacity": 1,
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 14,
                  "fontWeight": "600",
                  "letterSpacing": 0.75,
                },
                Object {
                  "textAlign": "center",
                },
                undefined,
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "hsl(211, 100%, 50%)",
                  },
                  undefined,
                  Object {
                    "color": "hsl(208, 7%, 46%)",
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            CONFIRM
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`Dialog component renders dialog to show single button in center 1`] = `
<Modal
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "hsla(0, 0%, 0%, 0.5)",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": 32,
      }
    }
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "hsl(0, 100%, 100%)",
          "borderRadius": 16,
          "marginHorizontal": 60,
          "maxWidth": 500,
          "paddingBottom": 36,
          "paddingHorizontal": 36,
          "paddingTop": 52,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 24,
              "fontWeight": "300",
              "letterSpacing": 0,
            },
            undefined,
            undefined,
            Object {
              "paddingBottom": 12,
            },
          ]
        }
      >
        bob
      </Text>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingTop": 52,
            },
            Object {
              "display": "none",
            },
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderWidth": 1,
              "color": "hsl(211, 100%, 50%)",
              "flexDirection": "row",
              "height": 36,
              "justifyContent": "center",
              "opacity": 1,
              "paddingHorizontal": 16,
              "width": "50%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 14,
                  "fontWeight": "600",
                  "letterSpacing": 0.75,
                },
                Object {
                  "textAlign": "center",
                },
                undefined,
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "hsl(211, 100%, 50%)",
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            CANCEL
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
